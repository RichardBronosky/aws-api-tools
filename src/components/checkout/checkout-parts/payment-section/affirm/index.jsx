import React from 'react'
import AffirmButton from './affirm-button'
import { getAffirmBody } from '../../../../../lib/helpers/checkout/payment-section/affirm'

export default class AffirmButtonWrapper extends React.Component {
  state = {
    affirmLoaded: false,
  }

  affirmFunc = (l, g, m, e, a, f, b) => {
    var d,
      c = l[m] || {},
      h = document.createElement(f),
      n = document.getElementsByTagName(f)[0],
      k = function(a, b, c) {
        return function() {
          a[b]._.push([c, arguments])
        }
      }
    c[e] = k(c, e, 'set')
    d = c[e]
    c[a] = {}
    c[a]._ = []
    d._ = []
    c[a][b] = k(c, a, b)
    a = 0
    for (b = 'set add save post open empty reset on off trigger ready setProduct'.split(' '); a < b.length; a++)
      d[b[a]] = k(c, e, b[a])
    a = 0
    for (b = ['get', 'token', 'url', 'items']; a < b.length; a++) d[b[a]] = function() {}
    h.async = !0
    h.src = g[f]
    n.parentNode.insertBefore(h, n)
    delete g[f]
    d(g)
    l[m] = c
    this.setState({ affirmLoaded: true })
  }

  affirmClick = () => {
    affirm.checkout(getAffirmBody())
    affirm.checkout.open()
  }

  render() {
    const _affirm_config = {
      public_api_key: process.env.GATSBY_AFFIRM_API_KEY,
      script: `${ process.env.GATSBY_AFFIRM_URL }/js/v2/affirm.js`,
    }
    return (
      <AffirmButton
        affirmLoaded={ this.state.affirmLoaded }
        affirmFunc={ this.affirmFunc }
        affirm_config={ _affirm_config }
        affirmClick={ this.affirmClick }
      />
    )
  }
}
