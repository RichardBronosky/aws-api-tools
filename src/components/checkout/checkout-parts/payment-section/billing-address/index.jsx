import React from 'react'
import BillingAddress from './billing-address'
import {
  validateBillingAddress,
  submitBillingAddress,
} from '../../../../../lib/helpers/checkout/payment-section/billing-address'

export default class BillingAddressWrapper extends React.Component {
  state = {
    invalidFields: [],
    loading: false,
  }

  submitBillingAddress = async () => {
    this.setBillingState({ loading: true })
    const invalidFields = await validateBillingAddress(this.props.order)
    this.setState({ invalidFields })
    if (invalidFields.length < 1) {
      submitBillingAddress(this.setBillingState)
    } else {
      this.setBillingState({ loading: false })
    }
  }

  setBillingState = state => this.setState(state)

  clearInvalidFields = () => this.setState({ invalidFields: [] })

  render() {
    const { order } = this.props
    const { invalidFields, loading } = this.state
    return (
      <BillingAddress
        order={ order }
        invalidFields={ invalidFields }
        loading={ loading }
        submitBillingAddress={ this.submitBillingAddress }
        clearInvalidFields={ this.clearInvalidFields }
        setBillingState={ this.setBillingState }
      />
    )
  }
}
