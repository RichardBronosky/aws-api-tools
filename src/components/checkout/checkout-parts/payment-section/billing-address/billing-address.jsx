import React from 'react'
import classNames from 'classnames'
import CheckoutInput from '../../checkout-input'
import ErrorMessage from '../../checkout-error-message'
import AddressSuggestionModal from '../../address-suggestion-modal'
import { states } from '../../../../../lib/helpers/geo-location'
import { setPayerInfo } from '../../../../../lib/helpers/checkout/payment-section/payment-section'
import {
  setBillingAddressInfo,
  onBillingStateChange,
  onChangeBillingDifferent,
} from '../../../../../lib/helpers/checkout/payment-section/billing-address'
import '../../../../../assets/css/components/checkout/checkout-parts/payment-section/billing-address.sass'
import loaderLight from '../../../../../assets/images/loader-light.svg'

export default ({ order, invalidFields, loading, submitBillingAddress, clearInvalidFields, setBillingState }) => {
  const suggestModal = invalidFields.length === 1 && invalidFields.includes('unable to verify')
  return (
    <div className="billing-container">
      <input
        type="checkbox"
        className="billing-address-checkbox"
        name="Edit Billing Address"
        label="Edit Billing Address"
        aria-label="Edit Billing Address"
        checked={ !order.payer.billingDifferent }
        onChange={ e => onChangeBillingDifferent(e, order) }
      />
      Billing address same as shipping
      { order.payer.billingDifferent && !order.payer.billingSubmitted && (
        <div className="billing-address">
          { invalidFields.length > 0 && (
            <ErrorMessage
              invalidFields={ invalidFields }
              customMessage={ suggestModal ? 'We were unable to verify your address. Please review your entry and try again.' : null }
            />
          ) }
          { suggestModal && (
            <AddressSuggestionModal
              setBillingState={ setBillingState }
              modalOpen={ suggestModal }
              order={ order }
              suggestion={ order.suggestedAddress }
              closeModal={ clearInvalidFields }
            />
          ) }
          <CheckoutInput
            type="text"
            field="firstName"
            label="First Name"
            info={ order.payer }
            setInfo={ setPayerInfo }
            invalidFields={ invalidFields }
            required
          />
          <CheckoutInput
            type="text"
            field="lastName"
            label="Last Name"
            info={ order.payer }
            setInfo={ setPayerInfo }
            invalidFields={ invalidFields }
            required
          />
          <CheckoutInput
            type="text"
            className="street"
            field="address1"
            label="Street Address"
            info={ order.billingAddress }
            setInfo={ setBillingAddressInfo }
            invalidFields={ invalidFields }
            required
          />
          <CheckoutInput
            type="text"
            className="apt"
            field="address2"
            label="Apt, Suite, Etc"
            info={ order.billingAddress }
            setInfo={ setBillingAddressInfo }
          />
          <CheckoutInput
            type="text"
            className="city"
            field="city"
            label="City"
            info={ order.billingAddress }
            setInfo={ setBillingAddressInfo }
            invalidFields={ invalidFields }
            required
          />
          <label className="label state" htmlFor="state">
            <p>State*</p>
            <select
              name="state"
              aria-label="State*"
              className={ classNames('state', {
                invalid: invalidFields.includes('state'),
              }) }
              value={ order.billingAddress.state.toUpperCase() }
              onChange={ selected => onBillingStateChange(selected, setBillingAddressInfo) }
            >
              { (!order.billingAddress.state || order.billingAddress.state === '') && (
                <option value="none">*State</option>
              ) }
              { states
                .filter(state => state[1] !== 'PR')
                .map(state => (
                  <option key={ state[1] } value={ state[1] }>
                    { state[1] }
                  </option>
                )) }
            </select>
          </label>
          <CheckoutInput
            type="text"
            className="zip"
            field="zip"
            label="Zip"
            info={ order.billingAddress }
            setInfo={ setBillingAddressInfo }
            invalidFields={ invalidFields }
            required
          />
          <div className="submit-billing-container">
            <button
              className="submit-billing-btn"
              tabIndex="0"
              value="Submit"
              aria-label="Submit"
              onClick={ () => submitBillingAddress() }
            >
              { !loading && <>Submit</> }
              { loading && <img className="loader" alt="Submitting new billing address" src={ loaderLight } /> }
            </button>
          </div>
        </div>
      ) }
      { order.payer.billingDifferent && order.payer.billingSubmitted && (
        <div className="billing">
          <h4>Billing Information</h4>(
          <button
            className="edit-billing-btn"
            tabIndex="0"
            value="Edit Billing Address"
            aria-label="Edit Billing Address"
            onClick={ () => setPayerInfo(false, 'billingSubmitted') }
          >
            Edit Billing Address
          </button>
          )
          <div className="left-info">
            { order.payer.firstName + ' ' + order.payer.lastName }
            <br />
            { order.billingAddress.address1 }
            { order.billingAddress.address2 !== '' && ' ' + order.billingAddress.address2 }
            <br />
            { order.billingAddress.city + ' ' }
            { order.billingAddress.state + ' ' }
            { order.billingAddress.zip }
          </div>
        </div>
      ) }
    </div>
  )
}
