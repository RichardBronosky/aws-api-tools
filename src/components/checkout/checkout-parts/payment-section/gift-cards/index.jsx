import React from 'react'
import GiftCards from './gift-cards'
import { getGiftCards } from '../../../../../lib/helpers/checkout/payment-section/gift-cards'

export default class GiftCardsWrapper extends React.Component {
  state = {
    unableToAddMessage: '',
    loading: false,
    invalidFields: [],
    removing: false,
  }

  setGiftCardState = (info, field, callback = null) => {
    if (field) {
      this.setState(
        {
          ...this.state,
          [field]: info,
        },
        callback
      )
    } else {
      this.setState(
        {
          ...this.state,
          ...info,
        },
        callback
      )
    }
  }

  render() {
    const { order } = this.props
    const { removing, loading, invalidFields, unableToAddMessage } = this.state
    let paymentProperties = false
    if (order && order.paymentInfo && order.paymentInfo.length > 0) {
      paymentProperties = true
    }
    const giftCards = getGiftCards()
    return (
      <GiftCards
        order={ order }
        removing={ removing }
        loading={ loading }
        invalidFields={ invalidFields }
        giftCards={ giftCards }
        paymentProperties={ paymentProperties }
        setGiftCardState={ this.setGiftCardState }
        unableToAddMessage={ unableToAddMessage }
      />
    )
  }
}
