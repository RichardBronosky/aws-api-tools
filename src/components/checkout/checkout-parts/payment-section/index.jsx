import React from 'react'
import { connect } from 'react-redux'
import { handleAffirm } from '../../../../lib/helpers/checkout/payment-section/affirm'
import PaymentSection from './payment-section'
import '../../../../assets/css/components/checkout/checkout-parts/payment-section/payment-section.sass'

class PaymentSectionWrapper extends React.Component {
  state = {
    morePaymentMethods: false,
    ccForm: null,
    mountLoad: false,
  }

  componentDidMount() {
    handleAffirm()
  }

  render() {
    const { paymentInvalidFields } = this.props
    return <PaymentSection { ...this.props } invalidFields={ paymentInvalidFields } />
  }
}

const mapStateToProps = state => {
  return {
    order: state.checkout.order,
    checkoutStep: state.checkout.checkoutStep,
    checkoutStepsCompleted: state.checkout.checkoutStepsCompleted,
    paymentInvalidFields: state.checkout.paymentInvalidFields,
  }
}

export default connect(mapStateToProps)(PaymentSectionWrapper)
