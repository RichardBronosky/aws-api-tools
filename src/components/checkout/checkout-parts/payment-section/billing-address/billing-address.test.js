import React from 'react'
import renderer from 'react-test-renderer'
import BillingAddress from './billing-address'
import * as paymentSectionHelper from '../../../../../lib/helpers/checkout/payment-section/payment-section'
import * as billingAddressHelper from '../../../../../lib/helpers/checkout/payment-section/billing-address'
import { order1 } from '../../../../../lib/mocks/checkoutDataMocks'

describe('BillingAddress', () => {
  let testProps = {
    order: {
      ...order1,
      payer: {
        ...order1.payer,
        billingDifferent: true,
        billingSubmitted: false,
      },
    },
    invalidFields: [],
    loading: false,
    submitBillingAddress: jest.fn(),
  }

  it('renders correctly with when user selects billing different and has not submitted', () => {
    const tree = renderer.create(<BillingAddress { ...testProps } />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with when user selects billing different and has submitted', () => {
    const tree = renderer
      .create(
        <BillingAddress
          { ...{ ...testProps, order: { ...order1, payer: { ...order1.payer, billingSubmitted: true } } } }
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with when user selects billing not different and has not submitted', () => {
    const tree = renderer
      .create(
        <BillingAddress
          { ...{ ...testProps, order: { ...order1, payer: { ...order1.payer, billingDifferent: false } } } }
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly while loading', () => {
    const tree = renderer.create(<BillingAddress { ...{ ...testProps, loading: true } } />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with invalid fields', () => {
    const tree = renderer.create(<BillingAddress { ...{ ...testProps, invalidFields: ['firstName'] } } />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when billing submitted', () => {
    const tree = renderer
      .create(
        <BillingAddress
          { ...{ ...testProps, order: { ...order1, payer: { ...order1.payer, billingSubmitted: true } } } }
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('calls onChangeBillingDifferent when simulating billing address different checkbox', () => {
    billingAddressHelper.onChangeBillingDifferent = jest.fn()
    const tree = renderer.create(<BillingAddress { ...testProps } />)
    tree.root.findByProps({ className: 'billing-address-checkbox' }).props.onChange()
    expect(billingAddressHelper.onChangeBillingDifferent).toHaveBeenCalled()
  })

  it('calls onBillingStateChange when simulating changing billing address state', () => {
    billingAddressHelper.onBillingStateChange = jest.fn()
    const tree = renderer.create(<BillingAddress { ...testProps } />)
    tree.root.findByProps({ className: 'state' }).props.onChange()
    expect(billingAddressHelper.onBillingStateChange).toHaveBeenCalled()
  })

  it('calls setPayerInfo when simulating click submit', () => {
    paymentSectionHelper.setPayerInfo = jest.fn()
    const tree = renderer.create(
      <BillingAddress { ...{ ...testProps, order: { ...order1, payer: { ...order1.payer, billingSubmitted: true } } } } />
    )
    tree.root.findByProps({ className: 'edit-billing-btn' }).props.onClick()
    expect(paymentSectionHelper.setPayerInfo).toHaveBeenCalled()
  })
})
