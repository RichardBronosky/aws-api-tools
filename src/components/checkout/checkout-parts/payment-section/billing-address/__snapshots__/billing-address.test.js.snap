// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BillingAddress renders correctly when billing submitted 1`] = `
<div
  className="billing-container"
>
  <input
    aria-label="Edit Billing Address"
    checked={false}
    className="billing-address-checkbox"
    label="Edit Billing Address"
    name="Edit Billing Address"
    onChange={[Function]}
    type="checkbox"
  />
  Billing address same as shipping
  <div
    className="billing"
  >
    <h4>
      Billing Information
    </h4>
    (
    <button
      aria-label="Edit Billing Address"
      className="edit-billing-btn"
      onClick={[Function]}
      tabIndex="0"
      value="Edit Billing Address"
    >
      Edit Billing Address
    </button>
    )
    <div
      className="left-info"
    >
      Tester Test
      <br />
      103 seffner ave
      <br />
      seffner 
      fl 
      33584
    </div>
  </div>
</div>
`;

exports[`BillingAddress renders correctly while loading 1`] = `
<div
  className="billing-container"
>
  <input
    aria-label="Edit Billing Address"
    checked={false}
    className="billing-address-checkbox"
    label="Edit Billing Address"
    name="Edit Billing Address"
    onChange={[Function]}
    type="checkbox"
  />
  Billing address same as shipping
  <div
    className="billing-address"
  >
    <label
      className="label firstName"
      htmlFor="firstName"
    >
      First Name*
       
      <input
        aria-invalid={false}
        aria-required={true}
        checked={false}
        id="firstName"
        name="First Name"
        onChange={[Function]}
        onClick={[Function]}
        placeholder="First Name"
        type="text"
        value="Tester"
      />
      <span />
    </label>
    <label
      className="label lastName"
      htmlFor="lastName"
    >
      Last Name*
       
      <input
        aria-invalid={false}
        aria-required={true}
        checked={false}
        id="lastName"
        name="Last Name"
        onChange={[Function]}
        onClick={[Function]}
        placeholder="Last Name"
        type="text"
        value="Test"
      />
      <span />
    </label>
    <label
      className="label address1"
      htmlFor="address1"
    >
      Street Address*
       
      <input
        aria-invalid={true}
        aria-required={true}
        checked={false}
        className="street"
        id="address1"
        name="Street Address"
        onChange={[Function]}
        onClick={[Function]}
        placeholder="Street Address"
        type="text"
        value="103 seffner ave"
      />
      <span />
    </label>
    <label
      className="label address2"
      htmlFor="address2"
    >
      Apt, Suite, Etc
       
      <input
        aria-invalid={null}
        checked={false}
        className="apt"
        id="address2"
        name="Apt, Suite, Etc"
        onChange={[Function]}
        onClick={[Function]}
        placeholder="Apt, Suite, Etc"
        type="text"
        value=""
      />
      <span />
    </label>
    <label
      className="label city"
      htmlFor="city"
    >
      City*
       
      <input
        aria-invalid={true}
        aria-required={true}
        checked={false}
        className="city"
        id="city"
        name="City"
        onChange={[Function]}
        onClick={[Function]}
        placeholder="City"
        type="text"
        value="seffner"
      />
      <span />
    </label>
    <label
      className="label state"
      htmlFor="state"
    >
      <p>
        State*
      </p>
      <select
        aria-label="State*"
        className="state"
        name="state"
        onChange={[Function]}
        value="FL"
      >
        <option
          value="AZ"
        >
          AZ
        </option>
        <option
          value="AL"
        >
          AL
        </option>
        <option
          value="AK"
        >
          AK
        </option>
        <option
          value="AR"
        >
          AR
        </option>
        <option
          value="CA"
        >
          CA
        </option>
        <option
          value="CO"
        >
          CO
        </option>
        <option
          value="CT"
        >
          CT
        </option>
        <option
          value="DC"
        >
          DC
        </option>
        <option
          value="DE"
        >
          DE
        </option>
        <option
          value="FL"
        >
          FL
        </option>
        <option
          value="GA"
        >
          GA
        </option>
        <option
          value="HI"
        >
          HI
        </option>
        <option
          value="ID"
        >
          ID
        </option>
        <option
          value="IL"
        >
          IL
        </option>
        <option
          value="IN"
        >
          IN
        </option>
        <option
          value="IA"
        >
          IA
        </option>
        <option
          value="KS"
        >
          KS
        </option>
        <option
          value="KY"
        >
          KY
        </option>
        <option
          value="LA"
        >
          LA
        </option>
        <option
          value="ME"
        >
          ME
        </option>
        <option
          value="MD"
        >
          MD
        </option>
        <option
          value="MA"
        >
          MA
        </option>
        <option
          value="MI"
        >
          MI
        </option>
        <option
          value="MN"
        >
          MN
        </option>
        <option
          value="MS"
        >
          MS
        </option>
        <option
          value="MO"
        >
          MO
        </option>
        <option
          value="MT"
        >
          MT
        </option>
        <option
          value="NE"
        >
          NE
        </option>
        <option
          value="NV"
        >
          NV
        </option>
        <option
          value="NH"
        >
          NH
        </option>
        <option
          value="NJ"
        >
          NJ
        </option>
        <option
          value="NM"
        >
          NM
        </option>
        <option
          value="NY"
        >
          NY
        </option>
        <option
          value="NC"
        >
          NC
        </option>
        <option
          value="ND"
        >
          ND
        </option>
        <option
          value="OH"
        >
          OH
        </option>
        <option
          value="OK"
        >
          OK
        </option>
        <option
          value="OR"
        >
          OR
        </option>
        <option
          value="PA"
        >
          PA
        </option>
        <option
          value="RI"
        >
          RI
        </option>
        <option
          value="SC"
        >
          SC
        </option>
        <option
          value="SD"
        >
          SD
        </option>
        <option
          value="TN"
        >
          TN
        </option>
        <option
          value="TX"
        >
          TX
        </option>
        <option
          value="UT"
        >
          UT
        </option>
        <option
          value="VT"
        >
          VT
        </option>
        <option
          value="VA"
        >
          VA
        </option>
        <option
          value="WA"
        >
          WA
        </option>
        <option
          value="WV"
        >
          WV
        </option>
        <option
          value="WI"
        >
          WI
        </option>
        <option
          value="WY"
        >
          WY
        </option>
      </select>
    </label>
    <label
      className="label zip"
      htmlFor="zip"
    >
      Zip*
       
      <input
        aria-invalid={true}
        aria-required={true}
        checked={false}
        className="zip"
        id="zip"
        name="Zip"
        onChange={[Function]}
        onClick={[Function]}
        placeholder="Zip"
        type="text"
        value="33584"
      />
      <span />
    </label>
    <div
      className="submit-billing-container"
    >
      <button
        aria-label="Submit"
        className="submit-billing-btn"
        onClick={[Function]}
        tabIndex="0"
        value="Submit"
      >
        <img
          alt="Submitting new billing address"
          className="loader"
          src="test-file-stub"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`BillingAddress renders correctly with invalid fields 1`] = `
<div
  className="billing-container"
>
  <input
    aria-label="Edit Billing Address"
    checked={false}
    className="billing-address-checkbox"
    label="Edit Billing Address"
    name="Edit Billing Address"
    onChange={[Function]}
    type="checkbox"
  />
  Billing address same as shipping
  <div
    className="billing-address"
  >
    <p
      className="invalid-message"
    >
      *Invalid
       
      first name
    </p>
    <label
      className="label firstName"
      htmlFor="firstName"
    >
      First Name*
       
      <input
        aria-invalid={true}
        aria-required={true}
        checked={false}
        className="invalid"
        id="firstName"
        name="First Name"
        onChange={[Function]}
        onClick={[Function]}
        placeholder="First Name"
        type="text"
        value="Tester"
      />
      <span />
    </label>
    <label
      className="label lastName"
      htmlFor="lastName"
    >
      Last Name*
       
      <input
        aria-invalid={false}
        aria-required={true}
        checked={false}
        id="lastName"
        name="Last Name"
        onChange={[Function]}
        onClick={[Function]}
        placeholder="Last Name"
        type="text"
        value="Test"
      />
      <span />
    </label>
    <label
      className="label address1"
      htmlFor="address1"
    >
      Street Address*
       
      <input
        aria-invalid={true}
        aria-required={true}
        checked={false}
        className="street"
        id="address1"
        name="Street Address"
        onChange={[Function]}
        onClick={[Function]}
        placeholder="Street Address"
        type="text"
        value="103 seffner ave"
      />
      <span />
    </label>
    <label
      className="label address2"
      htmlFor="address2"
    >
      Apt, Suite, Etc
       
      <input
        aria-invalid={null}
        checked={false}
        className="apt"
        id="address2"
        name="Apt, Suite, Etc"
        onChange={[Function]}
        onClick={[Function]}
        placeholder="Apt, Suite, Etc"
        type="text"
        value=""
      />
      <span />
    </label>
    <label
      className="label city"
      htmlFor="city"
    >
      City*
       
      <input
        aria-invalid={true}
        aria-required={true}
        checked={false}
        className="city"
        id="city"
        name="City"
        onChange={[Function]}
        onClick={[Function]}
        placeholder="City"
        type="text"
        value="seffner"
      />
      <span />
    </label>
    <label
      className="label state"
      htmlFor="state"
    >
      <p>
        State*
      </p>
      <select
        aria-label="State*"
        className="state"
        name="state"
        onChange={[Function]}
        value="FL"
      >
        <option
          value="AZ"
        >
          AZ
        </option>
        <option
          value="AL"
        >
          AL
        </option>
        <option
          value="AK"
        >
          AK
        </option>
        <option
          value="AR"
        >
          AR
        </option>
        <option
          value="CA"
        >
          CA
        </option>
        <option
          value="CO"
        >
          CO
        </option>
        <option
          value="CT"
        >
          CT
        </option>
        <option
          value="DC"
        >
          DC
        </option>
        <option
          value="DE"
        >
          DE
        </option>
        <option
          value="FL"
        >
          FL
        </option>
        <option
          value="GA"
        >
          GA
        </option>
        <option
          value="HI"
        >
          HI
        </option>
        <option
          value="ID"
        >
          ID
        </option>
        <option
          value="IL"
        >
          IL
        </option>
        <option
          value="IN"
        >
          IN
        </option>
        <option
          value="IA"
        >
          IA
        </option>
        <option
          value="KS"
        >
          KS
        </option>
        <option
          value="KY"
        >
          KY
        </option>
        <option
          value="LA"
        >
          LA
        </option>
        <option
          value="ME"
        >
          ME
        </option>
        <option
          value="MD"
        >
          MD
        </option>
        <option
          value="MA"
        >
          MA
        </option>
        <option
          value="MI"
        >
          MI
        </option>
        <option
          value="MN"
        >
          MN
        </option>
        <option
          value="MS"
        >
          MS
        </option>
        <option
          value="MO"
        >
          MO
        </option>
        <option
          value="MT"
        >
          MT
        </option>
        <option
          value="NE"
        >
          NE
        </option>
        <option
          value="NV"
        >
          NV
        </option>
        <option
          value="NH"
        >
          NH
        </option>
        <option
          value="NJ"
        >
          NJ
        </option>
        <option
          value="NM"
        >
          NM
        </option>
        <option
          value="NY"
        >
          NY
        </option>
        <option
          value="NC"
        >
          NC
        </option>
        <option
          value="ND"
        >
          ND
        </option>
        <option
          value="OH"
        >
          OH
        </option>
        <option
          value="OK"
        >
          OK
        </option>
        <option
          value="OR"
        >
          OR
        </option>
        <option
          value="PA"
        >
          PA
        </option>
        <option
          value="RI"
        >
          RI
        </option>
        <option
          value="SC"
        >
          SC
        </option>
        <option
          value="SD"
        >
          SD
        </option>
        <option
          value="TN"
        >
          TN
        </option>
        <option
          value="TX"
        >
          TX
        </option>
        <option
          value="UT"
        >
          UT
        </option>
        <option
          value="VT"
        >
          VT
        </option>
        <option
          value="VA"
        >
          VA
        </option>
        <option
          value="WA"
        >
          WA
        </option>
        <option
          value="WV"
        >
          WV
        </option>
        <option
          value="WI"
        >
          WI
        </option>
        <option
          value="WY"
        >
          WY
        </option>
      </select>
    </label>
    <label
      className="label zip"
      htmlFor="zip"
    >
      Zip*
       
      <input
        aria-invalid={true}
        aria-required={true}
        checked={false}
        className="zip"
        id="zip"
        name="Zip"
        onChange={[Function]}
        onClick={[Function]}
        placeholder="Zip"
        type="text"
        value="33584"
      />
      <span />
    </label>
    <div
      className="submit-billing-container"
    >
      <button
        aria-label="Submit"
        className="submit-billing-btn"
        onClick={[Function]}
        tabIndex="0"
        value="Submit"
      >
        Submit
      </button>
    </div>
  </div>
</div>
`;

exports[`BillingAddress renders correctly with when user selects billing different and has not submitted 1`] = `
<div
  className="billing-container"
>
  <input
    aria-label="Edit Billing Address"
    checked={false}
    className="billing-address-checkbox"
    label="Edit Billing Address"
    name="Edit Billing Address"
    onChange={[Function]}
    type="checkbox"
  />
  Billing address same as shipping
  <div
    className="billing-address"
  >
    <label
      className="label firstName"
      htmlFor="firstName"
    >
      First Name*
       
      <input
        aria-invalid={false}
        aria-required={true}
        checked={false}
        id="firstName"
        name="First Name"
        onChange={[Function]}
        onClick={[Function]}
        placeholder="First Name"
        type="text"
        value="Tester"
      />
      <span />
    </label>
    <label
      className="label lastName"
      htmlFor="lastName"
    >
      Last Name*
       
      <input
        aria-invalid={false}
        aria-required={true}
        checked={false}
        id="lastName"
        name="Last Name"
        onChange={[Function]}
        onClick={[Function]}
        placeholder="Last Name"
        type="text"
        value="Test"
      />
      <span />
    </label>
    <label
      className="label address1"
      htmlFor="address1"
    >
      Street Address*
       
      <input
        aria-invalid={true}
        aria-required={true}
        checked={false}
        className="street"
        id="address1"
        name="Street Address"
        onChange={[Function]}
        onClick={[Function]}
        placeholder="Street Address"
        type="text"
        value="103 seffner ave"
      />
      <span />
    </label>
    <label
      className="label address2"
      htmlFor="address2"
    >
      Apt, Suite, Etc
       
      <input
        aria-invalid={null}
        checked={false}
        className="apt"
        id="address2"
        name="Apt, Suite, Etc"
        onChange={[Function]}
        onClick={[Function]}
        placeholder="Apt, Suite, Etc"
        type="text"
        value=""
      />
      <span />
    </label>
    <label
      className="label city"
      htmlFor="city"
    >
      City*
       
      <input
        aria-invalid={true}
        aria-required={true}
        checked={false}
        className="city"
        id="city"
        name="City"
        onChange={[Function]}
        onClick={[Function]}
        placeholder="City"
        type="text"
        value="seffner"
      />
      <span />
    </label>
    <label
      className="label state"
      htmlFor="state"
    >
      <p>
        State*
      </p>
      <select
        aria-label="State*"
        className="state"
        name="state"
        onChange={[Function]}
        value="FL"
      >
        <option
          value="AZ"
        >
          AZ
        </option>
        <option
          value="AL"
        >
          AL
        </option>
        <option
          value="AK"
        >
          AK
        </option>
        <option
          value="AR"
        >
          AR
        </option>
        <option
          value="CA"
        >
          CA
        </option>
        <option
          value="CO"
        >
          CO
        </option>
        <option
          value="CT"
        >
          CT
        </option>
        <option
          value="DC"
        >
          DC
        </option>
        <option
          value="DE"
        >
          DE
        </option>
        <option
          value="FL"
        >
          FL
        </option>
        <option
          value="GA"
        >
          GA
        </option>
        <option
          value="HI"
        >
          HI
        </option>
        <option
          value="ID"
        >
          ID
        </option>
        <option
          value="IL"
        >
          IL
        </option>
        <option
          value="IN"
        >
          IN
        </option>
        <option
          value="IA"
        >
          IA
        </option>
        <option
          value="KS"
        >
          KS
        </option>
        <option
          value="KY"
        >
          KY
        </option>
        <option
          value="LA"
        >
          LA
        </option>
        <option
          value="ME"
        >
          ME
        </option>
        <option
          value="MD"
        >
          MD
        </option>
        <option
          value="MA"
        >
          MA
        </option>
        <option
          value="MI"
        >
          MI
        </option>
        <option
          value="MN"
        >
          MN
        </option>
        <option
          value="MS"
        >
          MS
        </option>
        <option
          value="MO"
        >
          MO
        </option>
        <option
          value="MT"
        >
          MT
        </option>
        <option
          value="NE"
        >
          NE
        </option>
        <option
          value="NV"
        >
          NV
        </option>
        <option
          value="NH"
        >
          NH
        </option>
        <option
          value="NJ"
        >
          NJ
        </option>
        <option
          value="NM"
        >
          NM
        </option>
        <option
          value="NY"
        >
          NY
        </option>
        <option
          value="NC"
        >
          NC
        </option>
        <option
          value="ND"
        >
          ND
        </option>
        <option
          value="OH"
        >
          OH
        </option>
        <option
          value="OK"
        >
          OK
        </option>
        <option
          value="OR"
        >
          OR
        </option>
        <option
          value="PA"
        >
          PA
        </option>
        <option
          value="RI"
        >
          RI
        </option>
        <option
          value="SC"
        >
          SC
        </option>
        <option
          value="SD"
        >
          SD
        </option>
        <option
          value="TN"
        >
          TN
        </option>
        <option
          value="TX"
        >
          TX
        </option>
        <option
          value="UT"
        >
          UT
        </option>
        <option
          value="VT"
        >
          VT
        </option>
        <option
          value="VA"
        >
          VA
        </option>
        <option
          value="WA"
        >
          WA
        </option>
        <option
          value="WV"
        >
          WV
        </option>
        <option
          value="WI"
        >
          WI
        </option>
        <option
          value="WY"
        >
          WY
        </option>
      </select>
    </label>
    <label
      className="label zip"
      htmlFor="zip"
    >
      Zip*
       
      <input
        aria-invalid={true}
        aria-required={true}
        checked={false}
        className="zip"
        id="zip"
        name="Zip"
        onChange={[Function]}
        onClick={[Function]}
        placeholder="Zip"
        type="text"
        value="33584"
      />
      <span />
    </label>
    <div
      className="submit-billing-container"
    >
      <button
        aria-label="Submit"
        className="submit-billing-btn"
        onClick={[Function]}
        tabIndex="0"
        value="Submit"
      >
        Submit
      </button>
    </div>
  </div>
</div>
`;

exports[`BillingAddress renders correctly with when user selects billing different and has submitted 1`] = `
<div
  className="billing-container"
>
  <input
    aria-label="Edit Billing Address"
    checked={false}
    className="billing-address-checkbox"
    label="Edit Billing Address"
    name="Edit Billing Address"
    onChange={[Function]}
    type="checkbox"
  />
  Billing address same as shipping
  <div
    className="billing"
  >
    <h4>
      Billing Information
    </h4>
    (
    <button
      aria-label="Edit Billing Address"
      className="edit-billing-btn"
      onClick={[Function]}
      tabIndex="0"
      value="Edit Billing Address"
    >
      Edit Billing Address
    </button>
    )
    <div
      className="left-info"
    >
      Tester Test
      <br />
      103 seffner ave
      <br />
      seffner 
      fl 
      33584
    </div>
  </div>
</div>
`;

exports[`BillingAddress renders correctly with when user selects billing not different and has not submitted 1`] = `
<div
  className="billing-container"
>
  <input
    aria-label="Edit Billing Address"
    checked={true}
    className="billing-address-checkbox"
    label="Edit Billing Address"
    name="Edit Billing Address"
    onChange={[Function]}
    type="checkbox"
  />
  Billing address same as shipping
</div>
`;
